inject-java/src/main/java/io/micronaut/annotation/processing/TypeElementVisitorProcessor.java

```diff
@@ -55,10 +58,30 @@
 public class TypeElementVisitorProcessor extends AbstractInjectAnnotationProcessor {
 
     private boolean executed = false;
+    private Collection<TypeElementVisitor> typeElementVisitors;
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
+        super.init(processingEnv);
+        this.typeElementVisitors = findTypeElementVisitors();
+    }
 
     @Override
     public Set<String> getSupportedOptions() {
-        return Collections.singleton("org.gradle.annotation.processing.aggregating");
+        Stream<String> baseOption = Stream.of("org.gradle.annotation.processing.aggregating");
+        Stream<String> visitorsOptions = typeElementVisitors
+                .stream()
+                .map(TypeElementVisitor::getSupportedOptions)
+                .flatMap(Collection::stream);
+        Stream<String> visitorsAnnotationsOptions = typeElementVisitors
+                .stream()
+                .filter(tev -> tev.getClass().isAnnotationPresent(SupportedOptions.class))
+                .map(TypeElementVisitor::getClass)
+                .map(cls -> cls.getAnnotation(SupportedOptions.class))
+                .flatMap((SupportedOptions supportedOptions) -> Arrays.stream(supportedOptions.value()));
+        return Stream.of(baseOption, visitorsAnnotationsOptions, visitorsOptions)
+                .flatMap(Stream::distinct)
```

Comment:
You collect to a `Set`, so I don't think `distinct` is needed.

Reply:
Done!